tests/components/nest/sensor_sdm_test.py

```diff
@@ -0,0 +1,248 @@
+    await hass.async_block_till_done()
+    return subscriber
+
+
+async def test_thermostat_device(hass):
+    """Test a thermostat with temperature and humidity sensors."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {
+                    "sdm.devices.traits.Info": {
+                        "customName": "My Sensor",
+                    },
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.1,
+                    },
+                    "sdm.devices.traits.Humidity": {
+                        "ambientHumidityPercent": 35.0,
+                    },
+                },
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is not None
+    assert temperature.state == "25.1"
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is not None
+    assert humidity.state == "35.0"
+
+
+async def test_no_devices(hass):
+    """Test no devices returned by the api."""
+    await setup_sensor(hass)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is None
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is None
+
+
+async def test_device_no_sensor_traits(hass):
+    """Test a device with applicable sensor traits."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {},
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.my_sensor_temperature")
+    assert temperature is None
+
+    humidity = hass.states.get("sensor.my_sensor_humidity")
+    assert humidity is None
+
+
+async def test_device_name_from_structure(hass):
+    """Test a device without a custom name, inferring name from structure."""
+    devices = {
+        "some-device-id": Device.MakeDevice(
+            {
+                "name": "some-device-id",
+                "type": "sdm.devices.types.Thermostat",
+                "traits": {
+                    "sdm.devices.traits.Temperature": {
+                        "ambientTemperatureCelsius": 25.2,
+                    },
+                },
+                "parentRelations": [
+                    {"parent": "some-structure-id", "displayName": "Some Room"}
+                ],
+            },
+            auth=None,
+        )
+    }
+    await setup_sensor(hass, devices)
+
+    temperature = hass.states.get("sensor.some_room_temperature")
+    assert temperature is not None
+    assert temperature.state == "25.2"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)